src/team_lib/mhba/xccl_mhba_lib.h

```diff
@@ -64,18 +67,24 @@ typedef struct xccl_mhba_context {
 typedef struct xccl_mhba_node {
     xccl_sbgp_t                  *sbgp;
     void                         *storage;
-    void                         *ctrl;
-    void                         *my_ctrl;
-    void                         *send_umr_data;
-    void                         *my_send_umr_data;
-    void                         *recv_umr_data;
-    void                         *my_recv_umr_data;
-    struct mlx5dv_mkey*          send_mkey;
-    struct mlx5dv_mkey*          recv_mkey;
-    int                          block_size; // todo fill
+    void                         *ctrl[MAX_CONCURRENT_OUTSTANDING_ALL2ALL];
```

Comment:
already said above, repeating here: array of structs instead of struct of arrays would be much more clean solution imho